hedera-node/src/main/java/com/hedera/services/state/submerkle/CurrencyAdjustments.java

```diff
@@ -115,32 +120,59 @@ public String toString() {
 	/* --- Helpers --- */
 	public TransferList toGrpc() {
 		var grpc = TransferList.newBuilder();
-		IntStream.range(0, hbars.length)
-				.mapToObj(i -> AccountAmount.newBuilder()
-						.setAmount(hbars[i])
-						.setAccountID(EntityIdUtils.asAccount(accountIds.get(i))))
-				.forEach(grpc::addAccountAmounts);
+		for (int i = 0; i < hbars.length; i++) {
+			grpc.addAccountAmounts(AccountAmount.newBuilder()
+					.setAmount(hbars[i])
+					.setAccountID(EntityNum.fromLong(accountNums[i]).toGrpcAccountId())
+					.build());
+		}
 		return grpc.build();
 	}
 
-	public static CurrencyAdjustments fromGrpc(TransferList grpc) {
-		return fromGrpc(grpc.getAccountAmountsList());
+	public static CurrencyAdjustments fromChanges(final long[] balanceChanges, final long[] changedAccounts,
```

Comment:
Just realized it will be clearer and safer if we only pass `Arrays.copyOfRange(` and `Arrays.copyOfRange(changedAccounts, numHbarBalancesChangedSoFar)` into this method (and the remove `touchedSoFar` argument).